### ISSUE TITLE
Support Multiple Base Collections

### ISSUE THREAD

**Issue by mcordingley** [jmorrell/backbone.obscura#9]
Hi there,

I've run into a situation where what I need is essentially an instance of Obscura, but one that merges multiple collections together. The specific use case is a combined list of "things" being fed into a Marionette CollectionView and collectively sorted by date.

I have a (currently broken) proof of concept at https://gist.github.com/mcordingley/7900358 based somewhat on some of your code.

It would be awesome if this sort of thing could make it into Obscura.

Thanks,
-Michael

**Comment by jmorrell** [comment]
I think this is out of scope for Obscura. I'd really like to keep it focused on solving the more-common filtering, sorting, paginating problem.

That said, this does sound like something useful that absolutely should exist, and the implementation wouldn't be too different from the way Obscura is written. With a quick look at your gist I think you're on the right track. You should be able to re-use the collection proxy module: https://github.com/jmorrell/backbone-collection-proxy and pull something together pretty quickly.

Once you have a MergedCollection, you should be able to pass that into Obscura with no problems if it satisfies the Backbone.Collection interface that Obscura relies on (methods + events), which it will if you use backbone-collection-proxy :) 

I'll try to review your gist later tonight. You might consider forking one of the Obscura sub-modules (e.g. https://github.com/jmorrell/backbone-paginated-collection) as a starting point? Reach out on email or twitter if you want advice or code review.

**Comment by jmorrell** [comment]
And I wrote Obscura as modularly as I did for just this type of use case :)

**Comment by mcordingley** [comment]
I did use your collection proxy in making this. :) I think I pulled the frame of the idea from the sorted collection instead of the paginated one, but the effect is the same.

I'm still doing some debugging and it may turn out that the merged collection is functioning correctly and that the problem is instead with my view. Regardless, a review would be much appreciated.

Also, thank you for responding so quickly and completely, and more so for the review offer.
